TST: Some jobs failed

Look like the scheduled CI logs. These jobs are failing:

* tests (macos-latest, 3.7, py37-test-sphinx20)
* tests (ubuntu-latest, 3.8, py38-test-sphinx24)
* tests (windows-latest, 3.8, py38-test-sphinx30)

```
__________________________________ test_conf ___________________________________

tmpdir = local('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_conf0')

    def test_conf(tmpdir):
    
        # Just make sure the docs build with the default sphinx-astropy configuration
    
        generate_files(tmpdir)
    
        src_dir = tmpdir.strpath
        html_dir = tmpdir.mkdir('html').strpath
    
        status = build_main(argv=['-W', '-b', 'html', src_dir, html_dir])
    
>       assert status == 0
E       assert 2 == 0

../../sphinx_astropy/tests/test_conf.py:54: AssertionError
----------------------------- Captured stdout call -----------------------------
Running Sphinx v2.0.1
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from /Users/runner/work/sphinx-astropy/sphinx-astropy/sphinx_astropy/local/python3_local_links.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/scipy/reference/objects.inv...
loading intersphinx inventory from https://matplotlib.org/objects.inv...
loading intersphinx inventory from https://docs.astropy.org/en/stable/objects.inv...
loading intersphinx inventory from https://docs.h5py.org/en/stable/objects.inv...
[autosummary] generating autosummary for: index.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
----------------------------- Captured stderr call -----------------------------

Exception occurred:
  File "/Users/runner/work/sphinx-astropy/sphinx-astropy/.tox/py37-test-sphinx20/lib/python3.7/site-packages/docutils/writers/html5_polyglot/__init__.py", line 445, in section_title_tags
    if (ids and self.settings.section_self_link
AttributeError: 'Values' object has no attribute 'section_self_link'
The full traceback has been saved in /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/sphinx-err-mrwsmz6q.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```
```
___________________________ test_intersphinx_toggle ____________________________

tmpdir = local('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/test_intersphinx_toggle0')
capsys = <_pytest.capture.CaptureFixture object at 0x109eb5950>

    def test_intersphinx_toggle(tmpdir, capsys):
    
        # Test the sphinx_astropy.ext.intersphinx_toggle extension
    
        generate_files(tmpdir)
    
        src_dir = tmpdir.strpath
        html_dir = tmpdir.mkdir('html').strpath
    
        status = build_main(argv=['-W', '-b', 'html', src_dir, html_dir, '-D', 'disable_intersphinx=1'])
    
>       assert status == 0
E       assert 2 == 0

../../sphinx_astropy/tests/test_conf.py:68: AssertionError
----------------------------- Captured stdout call -----------------------------
Running Sphinx v2.0.1
disabling intersphinx...
[autosummary] generating autosummary for: index.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
----------------------------- Captured stderr call -----------------------------

Exception occurred:
  File "/Users/runner/work/sphinx-astropy/sphinx-astropy/.tox/py37-test-sphinx20/lib/python3.7/site-packages/docutils/writers/html5_polyglot/__init__.py", line 445, in section_title_tags
    if (ids and self.settings.section_self_link
AttributeError: 'Values' object has no attribute 'section_self_link'
The full traceback has been saved in /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/sphinx-err-pt6q9apb.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

1. Do we still need to support Python 3.7 here?
2. Do we still need to support very old Sphinx versions here?

--- a/tox.ini
+++ b/tox.ini
@@ -15,8 +15,11 @@ deps =
     sphinx17: sphinx==1.7.*
     sphinx18: sphinx==1.8.*
     sphinx20: sphinx==2.0.*
+    sphinx20: docutils==0.17.*
     sphinx24: sphinx==2.4.*
+    sphinx24: docutils==0.17.*
     sphinx30: sphinx==3.0.*
+    sphinx30: docutils==0.17.*
     sphinx35: sphinx==3.5.*
     sphinx40: sphinx==4.0.*
     sphinxdev: git+https://github.com/sphinx-doc/sphinx#egg=sphinx

